src/services/inplayer.account.service.ts

```diff
@@ -44,36 +80,170 @@ export const logout = async () => {
 export const getUser = async (): Promise<Customer> => {
   try {
     const { data } = await InPlayer.Account.getAccountInfo();
-    return processInplayerAccount(data);
+    return processAccount(data);
   } catch {
     throw new Error('Failed to fetch user data.');
   }
 };
 
 export const getFreshJwtToken = async ({ auth }: { auth: AuthData }) => auth;
 
-// responsible to convert the InPlayer object to be compatible to the store
-function processInplayerAccount(account: AccountData): Customer {
+export const updateCustomer: UpdateCustomer = async (values) => {
+  try {
+    const fullName = `${values.firstName} ${values.lastName}`;
+
+    const response = await InPlayer.Account.updateAccount({
+      fullName,
+      metadata: {
+        firstName: values.firstName as string,
+        lastName: values.lastName as string,
+        ...(values?.consents && { consents: JSON.stringify(values.consents) }),
+      },
+    });
+
+    return {
+      errors: [],
+      // @ts-ignore
+      // wrong data type from InPlayer SDK (will be updated in the SDK)
+      responseData: processAccount(response.data),
+    };
+  } catch {
+    throw new Error('Failed to update user data.');
+  }
+};
+
+export const getPublisherConsents: GetPublisherConsents = async (config) => {
+  try {
+    const { inplayer } = config.integrations;
+    const { data } = await InPlayer.Account.getRegisterFields(inplayer?.clientId || '');
+
+    // @ts-ignore
+    // wrong data type from InPlayer SDK (will be updated in the SDK)
+    const result: Array<Consent> = data?.collection
```

Comment:
Maybe we should use a convention here to allow both consent and capture questions to be added?  Maybe fields that are consents could be named `consent.<something>` and then all other fields will be considered questions?

Reply:
Actually that is the case. This is the method that just retrieves them in the same format as cleeng's ones. We are storing them in our account metadata as : `consents: {jsonstring}`.
For capture we don't need anything special since only `firstName` and `lastName` are stored in the metadata.

So we have something like:

`customer:{
metadata:{
consents: {...},
firstName: ,
lastName:,
}
}`